utils/query_builder.py

```diff
@@ -53,21 +54,27 @@ def add_sql_limit_condition(self, limit):
         return self
 
     def add_sql_offset_condition(self, offset):
-        if offset:
+        if offset != None:
             self.offset = offset
         return self
 
     def add_sql_order_by_condition(self, attribute: str, order: Order):
         self.order_by = (attribute, order.name)
         return self
 
-
-    def add_sql_not_in_condition(self, attribute: str = None, ids_list: List[str] = None):
+    def add_sql_not_in_condition(
+        self, attribute: str = None, ids_list: List[str] = None
+    ):
         if ids_list and attribute and len(ids_list) > 0:
             ids_values = convert_list_to_tuple_string(ids_list)
             self.where_conditions.append(f"c.{attribute} NOT IN {ids_values}")
         return self
 
+    def add_date_range(self, date_range: dict = None):
+        if date_range:
+            self.date_range = date_range
+            return self
```

Comment:
return self either if condition is met or not.

Reply:
Ok, adjustments were done.